source/extensions/common/wasm/context.cc

```diff
@@ -1603,7 +1623,7 @@ Http::FilterHeadersStatus Context::decodeHeaders(Http::RequestHeaderMap& headers
 Http::FilterDataStatus Context::decodeData(Buffer::Instance& data, bool end_stream) {
   request_body_buffer_ = &data;
   end_of_stream_ = end_stream;
-  auto result = onRequestBody(data.length(), end_stream);
```

Comment:
Isn't this value still valid, even in the buffered case?

Reply:
If buffering was enabled, then we pass the length of the buffered data to the callback, and this value otherwise, so it's only accurate when buffering is disabled.

In the "on*Body" method we call "getBuffer" to figure out which one to get the length of. I figured that was cleaner than putting additional conditional logic in "on*Body" to determine whether to use this length or the length of the buffer from "getBuffer".

Reply:
Yeah, I saw that... But for some reason I assumed that the `decodeData()` call would have the buffered body buffer passed here, which turns out it's not the case. I agree that handling the extra login in `getBuffer` is cleaner.